py3-secretstorage.yaml

```diff
@@ -33,13 +35,38 @@ pipeline:
       repository: https://github.com/mitya57/secretstorage
       tag: ${{package.version}}
 
-  - name: Python Build
-    runs: |
-      python -m build
-      python3 -m installer -d "${{targets.destdir}}" dist/*.whl
-      find ${{targets.destdir}} -name "*.pyc" -exec rm -rf '{}' +
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provider-priority: ${{range.value}}
+      provides:
+        - py3-${{vars.pypi-package}}
+        - py3-secretstorage
+        - py3-SecretStorage # Needed until keyring can be updated
+      runtime:
+        - py${{range.key}}-cryptography
+        - py${{range.key}}-jeepney
+        - py${{range.key}}-cffi
+        - openssl
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+      - uses: strip
+    test:
+      pipeline:
+        - uses: python/import
+          with:
+            python: python${{range.key}}
+            import: secretstorage
 
-  - uses: strip
+test:
+  pipeline:
+    - uses: python/import
+      with:
+        import: secretstorage
```

Comment:
I'm more curious since the ranging over versions thing is somewhat new to me - but is this test pipeline needed? It looks like all the versions should already be tested in the subpackage test that does the import?

Reply:
It's not strictly speaking needed but it is helpful.  This file produces a package called `py3-secretstorage` that's empty and mostly useless.   I don't know how to stop this file from producing this package and there's always the possibility someone is using `py3-secretstorage` as a dep.  This test ensures that if you install `py3-secretstorage` you are given a package that actually installs the library and a workable python version.  So it's testing the `py3-secretstorage` metapackage/priority provider functionality rather than the `py3.XX-secretstorage` packages.  I don't specify a python version because it should only pull in a single version. If multiple are being pulled in to satisfy the deps it should fail and be investigated.